packages/date-time-picker/test/individual-picker-update-helpers.js

```diff
@@ -0,0 +1,94 @@
+import { sendKeys } from '@vaadin/test-runner-commands';
+import { nextRender, outsideClick } from '@vaadin/testing-helpers';
+
+async function triggerInputChange(trigger) {
+  if (trigger === 'outside click') {
+    outsideClick();
+  } else {
+    await sendKeys({ press: 'Enter' });
+  }
+}
+
+const inputChangeTriggers = ['outside click', 'enter'];
+
+export const clearValueOnFocusedInput = async (trigger) => {
+  await sendKeys({ press: 'ControlOrMeta+A' });
+  await sendKeys({ press: 'Backspace' });
+  await triggerInputChange(trigger);
+  await nextRender();
+};
+
+export const changeValueOnFocusedInput = async (newValue, trigger) => {
+  await sendKeys({ press: 'ControlOrMeta+A' });
+  await sendKeys({ type: newValue });
+  await triggerInputChange(trigger);
+  await nextRender();
+};
+
+export const getPicker = (dateTimePicker, pickerType) => {
+  return pickerType === 'date-picker' ? dateTimePicker.__datePicker : dateTimePicker.__timePicker;
+};
+
+export const getOtherPickerType = (pickerType) => {
+  return pickerType === 'date-picker' ? 'time-picker' : 'date-picker';
+};
+
+export const getPickerInitialValue = (pickerType) => {
+  return pickerType === 'date-picker' ? '2/2/2022' : '02:02';
+};
+
+export const getPickerNewValue = (pickerType) => {
+  return pickerType === 'date-picker' ? '1/1/2023' : '13:00';
+};
+
+export const changeStateOnFocusedInput = async (pickerType, newPickerState, trigger) => {
+  if (newPickerState === 'empty') {
+    await clearValueOnFocusedInput(trigger);
+  } else {
+    const newValue = newPickerState === 'unparsable' ? 'unparsableNew' : getPickerNewValue(pickerType);
+    await changeValueOnFocusedInput(newValue, trigger);
+  }
+};
+
+export const initializePickerState = async (dateTimePicker, pickerType, initialState) => {
+  if (initialState === 'empty') {
+    return;
+  }
+  const picker = getPicker(dateTimePicker, pickerType);
+  picker.focus();
+  const value = initialState === 'parsable' ? getPickerInitialValue(pickerType) : 'unparsableInitial';
+  await changeValueOnFocusedInput(value);
+  picker.blur();
+};
+
+export const initializePickerStates = async (
+  dateTimePicker,
+  pickerType,
+  pickerInitialState,
+  otherPickerInitialState,
+) => {
+  await initializePickerState(dateTimePicker, pickerType, pickerInitialState);
+  await initializePickerState(dateTimePicker, getOtherPickerType(pickerType), otherPickerInitialState);
+};
+
+export const individualPickerUpdateTestSetup = (individualPickerTest) => {
+  ['date-picker', 'time-picker'].forEach((pickerType) => {
```

Comment:
Would it be possible to simplify or restructure these tests a bit to improve readability? The deep nesting is making it tricky to understand the test cases. Maybe it would be easier to follow if the tests were more explicit, even with some duplication.

Reply:
I started with that approach, but it seemed to bloat the tests by quite a lot. But I don't have a strong opinion, and if you think it is hard to read I can change it. Maybe we can keep `['date-picker', 'time-picker'].forEach(...)` and explicitly add tests for different states&updates. What do you think?

Reply:
> Maybe we can keep ['date-picker', 'time-picker'].forEach(...) and explicitly add tests for different states&updates

Sounds good! Let's try this.

Reply:
Separated the tests.